File: cpp/tensorrt_llm/nanobind/process_group/bindings.cpp
/*
 * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 * SPDX-License-Identifier: Apache-2.0
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#include "bindings.h"
#include <nanobind/stl/string.h>

#include "tensorrt_llm/common/bindingUtils.h"
#include "tensorrt_llm/runtime/utils/pgUtils.h"

namespace nb = nanobind;

namespace tensorrt_llm::nanobind::process_group
{

void initBindings(nb::module_& m)
{

    m.def("init_pg",
        [](nb::object world_pg_obj, nb::object local_pg_obj, std::string const& pybind11_abi)
        {
            using Pg = c10d::ProcessGroup;
            using E = nb::python_error;

            pg_utils::init_pg(common::get_intrusive_ptr<Pg, E>(world_pg_obj.ptr(), pybind11_abi),
                common::get_intrusive_ptr<Pg, E>(local_pg_obj.ptr(), pybind11_abi));
        });
}

} // namespace tensorrt_llm::nanobind::process_group

File: tensorrt_llm/grpc/smg/bindings.py
# SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
# SPDX-License-Identifier: Apache-2.0
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""Generated protobuf bindings used by the SMG adapter."""

try:
    from smg_grpc_proto.generated import trtllm_service_pb2, trtllm_service_pb2_grpc
except ModuleNotFoundError as e:
    if e.name != "smg_grpc_proto":
        raise
    raise ModuleNotFoundError(
        "The SMG gRPC adapter requires the optional 'smg-grpc-proto' package, "
        "which is not part of the default TensorRT-LLM installation. Install it "
        'with: pip install "tensorrt_llm[grpc-smg]"',
        name=e.name,
    ) from e

__all__ = ["trtllm_service_pb2", "trtllm_service_pb2_grpc"]
